api/v1alpha1/gatewayclass_types.go

```diff
@@ -61,6 +61,29 @@ type GatewayClassSpec struct {
 	// +required
 	Controller string `json:"controller" protobuf:"bytes,1,opt,name=controller"`
 
+	// GatewayNamespaceSelector is a selector of valid namespaces that Gateways
+	// can use this GatewayClass from. A nil or empty selector (default)
+	// indicates that Gateways can use this GatewayClass from any namespace.
+	//
+	// Support: Core
+	//
+	// +optional
+	GatewayNamespaceSelector *metav1.LabelSelector `json:"gatewayNamespaceSelector" protobuf:"bytes,2,opt,name=gatewayNamespaceSelector"`
```

Comment:
Change this to `AllowedGatewayNamespaces`. It should not be a pointer. Godoc should indicate that it is not a pointer because we don't want the nil behaviour (ie selects none). `omitempty` added to JSON tags.

Reply:
Updated, thanks!